### ISSUE TITLE
GUI for model optimization analysis

### ISSUE THREAD

**Issue by TheJoshBrod** [TheJoshBrod/Ghost-in-the-Shell-GPU-Compiler#4]
## Context

To make this project more accessible there should be a GUI tool to more direct control over model optimization.

Work has started on [GUI Branch](https://github.com/TheJoshBrod/CGinS/tree/gui)

## Required Features

- Project load/saving containing:
    - PyTorch code
    - Weights pt file
    - Directory location for validation set
    - Performance and CUDA files for all kernels
    - Model metadata
- API config containing:
    - api_key
    - model provider
    - model name
- Generate kernels of ops that don't have baseline version
- Optimize kernels at a per PyTorch object layer
- Export optimized model

---

Check comments below for example mock-ups

**Comment by TheJoshBrod** [comment]
Create new project

<img width="1370" height="984" alt="Image" src="https://github.com/user-attachments/assets/3faea07c-af87-42ff-a5aa-89243bb6cbde" />
